pkg/e2e/verify/routes.go

```diff
@@ -33,54 +27,48 @@ func init() {
 	alert.RegisterGinkgoAlert(routesTestName, "SD-CICD", "Diego Santamaria", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
 }
 
-var _ = ginkgo.Describe(routesTestName, ginkgo.Ordered, label.E2E, func() {
+var _ = ginkgo.Describe(routesTestName, ginkgo.Ordered, label.HyperShift, label.E2E, func() {
 	var h *helper.H
 	var client *resources.Resources
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
 
 	ginkgo.BeforeAll(func() {
 		h = helper.New()
 		client = h.AsUser("")
 	})
 
-	ginkgo.It("exist for console and is operational", label.HyperShift, func(ctx context.Context) {
-		var consoleRoute v1.Route
-		err := client.Get(ctx, consoleLabel, consoleNamespace, &consoleRoute)
-		expect.NoError(err)
-
-		for _, ingress := range consoleRoute.Status.Ingress {
-			response, err := validateRoute(ctx, ingress)
-			Expect(err).NotTo(HaveOccurred(), "failed retrieving %s site", consoleRoute.Name)
-			Expect(response.StatusCode).To(Equal(http.StatusOK))
+	ginkgo.DescribeTable("exists and is operational", func(ctx context.Context, name, namespace string) {
+		if name == "oauth-openshift" && viper.GetBool(config.Hypershift) {
+			ginkgo.Skip("OAuth route is not available on a HyperShift cluster")
 		}
-	})
 
-	ginkgo.It("exist for oauth and is operational", func(ctx context.Context) {
-		if viper.GetBool(config.Hypershift) {
-			ginkgo.Skip("OAuth route is not deployed to a ROSA hosted-cp cluster")
-		}
+		err := wait.For(func() (bool, error) {
+			err := client.Get(ctx, name, namespace, &routev1.Route{})
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return true, nil
+		}, wait.WithTimeout(3*time.Minute))
+		expect.NoError(err)
 
-		var oauthRoute v1.Route
-		err := client.Get(ctx, oauthName, oauthNamespace, &oauthRoute)
+		var route routev1.Route
+		err = client.Get(ctx, name, namespace, &route)
 		expect.NoError(err)
 
-		for _, ingress := range oauthRoute.Status.Ingress {
-			response, err := validateRoute(ctx, ingress)
-			Expect(err).NotTo(HaveOccurred(), "failed retrieving %s site", oauthRoute.Name)
-			Expect(response.StatusCode).To(Equal(http.StatusForbidden))
+		for _, ingress := range route.Status.Ingress {
+			response, err := httpClient.Get(fmt.Sprintf("https://%s", ingress.Host))
+			expect.NoError(err)
+			Expect(response.StatusCode).To(Equal(http.StatusOK))
```

Comment:
The HTTP status code differs between the console URL and OAuth URL. Console URL returns a `200` and OAuth URL returns `403`.

Can we update the `ginko.Entry` to pass in a fourth parameter for the http status and use that as part of the assertion?

```
•
------------------------------
• [FAILED] [6.789 seconds]
[Suite: e2e] Routes exists and is operational [It] OAuth [HyperShift, E2E]
/home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:72

  [FAILED] Expected
      <int>: 403
  to equal
      <int>: 200
  In [It] at: /home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:67 @ 01/13/23 12:44:38.818
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:
  [FAIL] [Suite: e2e] Routes exists and is operational [It] OAuth [HyperShift, E2E]
  /home/rywillia/osd/osde2e/pkg/e2e/verify/routes.go:67
```

Reply:
good catch, will update